### PR TITLE
implement delivered method of datatrust

### PR DESCRIPTION
**Pull request by robrobbins**
in which a datatrust can get paid.

### CODE REVIEW

File: contracts/Datatrust.vy [hunk 1]
```diff
@@ -258,3 +258,24 @@ def getAccessCredits(hash: bytes32) -> wei_value:
...
+  @param url A hash of the URL that the backend delivered to
+  """
+  assert msg.sender == self.backend_address
+  owner: address = self.deliveries[delivery].owner
+  delivered: uint256 = self.deliveries[delivery].bytes_delivered
+  assert delivered >= self.deliveries[delivery].bytes_requested
+  # clear the delivery record first
+  clear(self.deliveries[delivery])
+  # now pay the datatrust from the banked delivery request
+  back_fee: wei_value = (self.parameterizer.getCostPerByte() * delivered) / (100 / self.parameterizer.getBackendPayment())
```
Comment (rbharath) [inline contracts/Datatrust.vy]: Is there an edge failure case here when delivered > self.deliveries[delivery].bytes_requested? There may not be enough funds to pay.

Comment (robrobbins) [reply contracts/Datatrust.vy]: let's just adjust to this:

we'll pay the backend according to bytes_requested (but still making sure bytes_delivered is >= requested)

File: contracts/Datatrust.vy [hunk 2]
```diff
@@ -258,3 +258,25 @@ def getAccessCredits(hash: bytes32) -> wei_value:
...
+  @notice Allow a backend to collect its payment.
+  @dev We check that a backend has delivered, at least, the amount of bytes requested.
+  NOTE: bytes_requested is the multiplier for backend payment.
+  @param delivery Identifier of the delivery in question
+  @param url A hash of the URL that the backend delivered to
+  """
+  assert msg.sender == self.backend_address
+  owner: address = self.deliveries[delivery].owner
+  requested: uint256 = self.deliveries[delivery].bytes_requested
+  assert self.deliveries[delivery].bytes_delivered >= requested
```
Comment (robrobbins) [inline contracts/Datatrust.vy]: same check here

File: contracts/Datatrust.vy [hunk 3]
```diff
@@ -258,3 +258,25 @@ def getAccessCredits(hash: bytes32) -> wei_value:
...
+  # clear the delivery record first
+  clear(self.deliveries[delivery])
+  # now pay the datatrust from the banked delivery request
+  back_fee: wei_value = (self.parameterizer.getCostPerByte() * requested) / (100 / self.parameterizer.getBackendPayment())
```
Comment (robrobbins) [inline contracts/Datatrust.vy]: but only pay based on the request

**Pull request merged**